test/unittests/execute_test.cpp

```diff
@@ -320,6 +320,346 @@ TEST(execute, i64_load_trap)
+    instance.memory[0] = 0x00;
+    instance.memory[1] = 0x80;
+    instance.memory[2] = 0xf1;
+    const auto [trap, ret] = execute(instance, 0, {65537});
+
+    ASSERT_TRUE(trap);
+}
+
+TEST(execute, i64_load16_u)
+{
+    Module module;
+    module.memorysec.emplace_back(Memory{{1, 1}});
+    module.codesec.emplace_back(
+        Code{0, {Instr::local_get, Instr::i64_load16_u, Instr::end}, {0, 0, 0, 0, 0, 0, 0, 0}});
+
+    auto instance = instantiate(module);
+    instance.memory[0] = 0x01;
+    instance.memory[1] = 0x80;
+    instance.memory[2] = 0xf1;
+    const auto [trap, ret] = execute(instance, 0, {0});
+
+    ASSERT_FALSE(trap);
+    ASSERT_EQ(ret.size(), 1);
+    ASSERT_EQ(static_cast<uint64_t>(ret[0]), 32769);
+}
+
+TEST(execute, i64_load16_u_trap)
+{
+    Module module;
+    module.memorysec.emplace_back(Memory{{1, 1}});
+    module.codesec.emplace_back(
+        Code{0, {Instr::local_get, Instr::i64_load16_u, Instr::end}, {0, 0, 0, 0, 0, 0, 0, 0}});
+
+    auto instance = instantiate(module);
+    instance.memory[0] = 0x01;
+    instance.memory[1] = 0x80;
+    instance.memory[2] = 0xf1;
+    const auto [trap, ret] = execute(instance, 0, {65537});
+
+    ASSERT_TRUE(trap);
+}
+
+TEST(execute, i64_load32_s)
+{
+    Module module;
+    module.memorysec.emplace_back(Memory{{1, 1}});
+    module.codesec.emplace_back(
+        Code{0, {Instr::local_get, Instr::i64_load32_s, Instr::end}, {0, 0, 0, 0, 0, 0, 0, 0}});
+
+    auto instance = instantiate(module);
+    instance.memory[0] = 0x00;
+    instance.memory[1] = 0x00;
+    instance.memory[2] = 0x00;
+    instance.memory[3] = 0x80;
+    instance.memory[4] = 0xf1;
+    const auto [trap, ret] = execute(instance, 0, {0});
+
+    ASSERT_FALSE(trap);
+    ASSERT_EQ(ret.size(), 1);
+    ASSERT_EQ(static_cast<int64_t>(ret[0]), -2147483648);
+}
+
+TEST(execute, i64_load32_s_trap)
+{
+    Module module;
+    module.memorysec.emplace_back(Memory{{1, 1}});
+    module.codesec.emplace_back(
+        Code{0, {Instr::local_get, Instr::i64_load32_s, Instr::end}, {0, 0, 0, 0, 0, 0, 0, 0}});
+
+    auto instance = instantiate(module);
+    instance.memory[0] = 0x00;
+    instance.memory[1] = 0x00;
+    instance.memory[2] = 0x00;
+    instance.memory[3] = 0x80;
+    instance.memory[4] = 0xf1;
+    const auto [trap, ret] = execute(instance, 0, {65537});
+
+    ASSERT_TRUE(trap);
+}
+
+TEST(execute, i64_load32_u)
+{
+    Module module;
+    module.memorysec.emplace_back(Memory{{1, 1}});
+    module.codesec.emplace_back(
+        Code{0, {Instr::local_get, Instr::i64_load32_u, Instr::end}, {0, 0, 0, 0, 0, 0, 0, 0}});
+
+    auto instance = instantiate(module);
+    instance.memory[0] = 0x01;
+    instance.memory[1] = 0x00;
+    instance.memory[2] = 0x00;
+    instance.memory[3] = 0x80;
+    instance.memory[4] = 0xf1;
+    const auto [trap, ret] = execute(instance, 0, {0});
+
+    ASSERT_FALSE(trap);
+    ASSERT_EQ(ret.size(), 1);
+    ASSERT_EQ(static_cast<uint64_t>(ret[0]), 2147483649);
```

Comment:
I think it would be better to write expectation in hex, at least for unsigned tests.

Reply:
Also `ret[0]` is already `uint64_t`, casting is not needed